iceoryx_hoofs/source/posix_wrapper/thread.cpp

```diff
@@ -16,38 +16,97 @@
 
 #include "iceoryx_hoofs/posix_wrapper/thread.hpp"
 #include "iceoryx_hoofs/cxx/helplets.hpp"
+#include "iceoryx_hoofs/internal/log/hoofs_logging.hpp"
 #include "iceoryx_hoofs/posix_wrapper/posix_call.hpp"
 
 namespace iox
 {
 namespace posix
 {
-void setThreadName(pthread_t thread, const ThreadName_t& name) noexcept
+Thread::~Thread() noexcept
 {
-    posixCall(iox_pthread_setname_np)(thread, name.c_str()).successReturnValue(0).evaluate().or_else([](auto& r) {
-        // String length limit is ensured through cxx::string
-        // ERANGE (string too long) intentionally not handled to avoid untestable and dead code
-        std::cerr << "This should never happen! " << r.getHumanReadableErrnum() << std::endl;
-        cxx::Ensures(false && "internal logic error");
-    });
+    if (m_isJoinable)
+    {
+        auto joinResult = posixCall(pthread_join)(m_threadHandle, nullptr).successReturnValue(0).evaluate();
```

Comment:
Can you create a `#define NULLPTR_NULLPTR_NULLPTR_NULLPTR_NULLPTR nullptr;` and use it instead of `nullptr` in order to force clang-format to break the lines at the dot?